server/tserver/src/main/java/org/apache/accumulo/tserver/scan/LookupTask.java

```diff
@@ -87,23 +85,23 @@ public void run() {
       Key partNextKey = null;
       boolean partNextKeyInclusive = false;
 
-      Iterator<Entry<KeyExtent,List<Range>>> iter = session.queries.entrySet().iterator();
+      // Iterator<Entry<KeyExtent,List<Range>>> iter = session.queries.entrySet().iterator();
+      log.error("number of extents: " + session.queries.size() + " -- " + session.queries.keySet());
 
       // check the time so that the read ahead thread is not monopolized
-      while (iter.hasNext() && bytesAdded < maxResultsSize
+      while (!session.queries.isEmpty() && bytesAdded < maxResultsSize
           && (System.currentTimeMillis() - startTime) < maxScanTime) {
-        Entry<KeyExtent,List<Range>> entry = iter.next();
-
-        iter.remove();
+        KeyExtent extent = session.queries.keySet().iterator().next();
```

Comment:
Question: Does this really walk the iterator - or is it always returning the first entry pointed at by the iterator?

Reply:
Was just trying to get the first element, not iterating.  In any case, I modified the code to use a LinkedList.